src/mono/mono/mini/interp/transform.c

```diff
@@ -8467,6 +8467,24 @@ generate_compacted_code (TransformData *td)
 		InterpInst *ins = bb->first_ins;
 		bb->native_offset = GPTRDIFF_TO_INT (ip - td->new_code);
 		td->cbb = bb;
+
+#if HOST_BROWSER
+		if (bb->backwards_branch_target && rtm->contains_traces) {
+			if (!rtm->backward_branch_offsets) {
+				// Most methods in System.Runtime.Tests have 8 or fewer branch targets,
+				//  and many have more than 4, so this is a decent starting size
+				rtm->backward_branch_offsets_size = 8;
+				rtm->backward_branch_offsets = g_malloc0(sizeof(guint16) * rtm->backward_branch_offsets_size);
```

Comment:
Would be better to keep this in a local variable, and allocate it using mono_mem_manager_alloc0 () at the end of this function, most of the memory belonging to InterpMethod's is allocated that way.

Reply:
so like stackalloc a big enough scratch buffer, then copy it to a perfectly sized one allocated with the mem manager? that makes sense

Reply:
Its ok to malloc it and free it after copying it to the final memory as well.

Reply:
isn't that mempool freed at the end of compilation? the jiterp needs it after compilation

Reply:
td->mem_manager is not freed.